src/main/haskell/kore/src/Kore/ASTTraversals.hs

```diff
@@ -3,135 +3,35 @@ Module      : Kore.ASTTraversals
-        -> m (Either result ( Pattern level variable (Fix (pat variable))
-                            , m result -> m result
-                            )
-             )
-       )
-    -> (forall level . MetaOrObject level
-        => Pattern level variable result -> m result
-       )
-    -> (Fix (pat variable) -> m result)
-patternTopDownVisitorM preprocess postprocess =
-    fixTopDownVisitorM
-        fixPreprocess
-        (unifiedPatternApply postprocess)
-  where
-    {-
-    fixPreprocess
-        :: (pat' (Fix pat')
-        -> m (Either result ( pat' (Fix pat') , m result -> m result))
-        )
-      where pat' = pat variable
-    -}
-    fixPreprocess =
-        unifiedPatternApply
-            (preprocess
-            >=>
-            \case
-                Left r       -> return (Left r)
-                Right (p, t) -> return (Right (unifyPattern p, t))
-            )
-
-{-|'patternBottomUpVisitorM' is the specialization of 'patternTopDownVisitorM' where
-the preprocessor function always requests the recursive visitation of its
-children, basically resulting in a bottom-up visitor given by the aggregation
-function.
-
-The aggreagation function provided as argument is a local visitor/reducer
-which assumes that all children patterns of a pattern have been visited and
-transformed into results and aggregates these results into a new result.
--}
-patternBottomUpVisitorM
-    :: (Monad m, UnifiedPatternInterface pat, Traversable (pat variable))
-    => (forall level . MetaOrObject level
-        => Pattern level variable result -> m result)
-    -> (Fix (pat variable) -> m result)
-patternBottomUpVisitorM reduce =
-    fixBottomUpVisitorM (unifiedPatternApply reduce)
-
--- |'patternTopDownVisitor' is the non-monadic version of 'patternTopDownVisitorM'.
-patternTopDownVisitor
-    :: (UnifiedPatternInterface pat, Functor (pat variable))
-    => (forall level . MetaOrObject level
-        => Pattern level variable (Fix (pat variable))
-        -> Either result (Pattern level variable (Fix (pat variable)))
-       )
-    -> (forall level . MetaOrObject level
-        => Pattern level variable result -> result)
-    -> (Fix (pat variable) -> result)
-patternTopDownVisitor preprocess postprocess =
-    fixTopDownVisitor fixPreprocess fixPostprocess
-  where
-    fixPostprocess = unifiedPatternApply postprocess
-    fixPreprocess = unifiedPatternApply (fmap unifyPattern . preprocess)
+-- |'patternBottomUpVisitor' is @cata . unifiedPatternApply@
 
--- |'patternBottomUpVisitor' is the non-monadic version of 'patternBottomUpVisitorM'.
 patternBottomUpVisitor
     :: (UnifiedPatternInterface pat, Functor (pat variable))
     => (forall level . MetaOrObject level
         => Pattern level variable result -> result)
     -> (Fix (pat variable) -> result)
-patternBottomUpVisitor reduce = fixBottomUpVisitor (unifiedPatternApply reduce)
+patternBottomUpVisitor reduce = cata (unifiedPatternApply reduce)
```

Comment:
It appears that this module now defines exactly one function, which has a one line definition, is that right? Maybe we should move it into `Kore.AST.Common` with `unifiedPatternApply`.